src/audio/copier/copier.c

```diff
@@ -100,6 +101,41 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 	}
 
 	dev->pipeline = ipc_pipe->pipeline;
+	/* Calculation of the period in which the component should be scheduled based on
+	 * the input buffer size.
+	 */
+	if (copier->gtw_cfg.dma_buffer_size > 0 && dev->pipeline->long_period) {
+		/* Size of the single sample in bytes. */
+		uint32_t sample_size = copier->base.audio_fmt.channels_count *
+				       (copier->base.audio_fmt.depth / 8);
+		/* Size of the one second of audio data in bytes. */
+		uint32_t one_s = sample_size * copier->base.audio_fmt.sampling_frequency;
+		/* Size of the ten millisecond of audio data in bytes. */
+		uint32_t ten_ms = one_s / 100;
+		/* The number of ten milliseconds chunks of data that will fit into the buffer. */
+		uint32_t chunk_count = copier->gtw_cfg.dma_buffer_size / ten_ms;
+
+		/* If buffer can fit at more than ten milliseconds of data we can try to schedule
+		 * pipe on periods bigger than one millisecond.
+		 */
+		if (chunk_count) {
+			dev->deep_buffering = true;
+			if (chunk_count == 1)
+				dev->period = (5 * LL_TIMER_PERIOD_US);
+			else
+				dev->period = (10 * LL_TIMER_PERIOD_US);
+
```

Comment:
With so many inputs (both from runtime PCM config as well as topology), I think this warrants printing out the modified period size to FW log.